lib/src/yaml_edit/src/errors.dart

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// Error thrown when a function is passed an invalid path.
+class PathError extends ArgumentError {
+  /// The path that caused the error
+  final Iterable<Object> path;
+
+  /// The last element of [path] that could be traversed.
+  Object parentNode;
+
+  PathError(this.path, Object invalidKeyOrIndex, this.parentNode,
+      [String message])
+      : super.value(invalidKeyOrIndex, 'path', message);
+
+  PathError.unexpected(this.path, String message) : super(message);
```

Comment:
Any reason to make this public?

We could make private constructors... and create top-level functions like `createPathError` for creating instances..
Then in `yaml_edit.dart` we can do `export 'src/errors.dart' show PathError, AliasError;`, so we hide the constructor for end-users..

Reply:
hm... what would the reason to export the errors be, if we're not expecting end users to catch them :O

Reply:
Correct, indeed some people would argue that we need not export it... I would be okay with that too.

We shouldn't reference it by name in the documentation comments, elsewhere then... but we can just reference the super class instead (as it's from the Dart-SDK)...

Reply:
I changed it such that the documentation comments for the public methods use `[ArgumentError]`, and the private methods say `[PathError]`!